Made it run in Matlab R2015b

DennisEckmeier opened the pull request:
Hi,

I'm the new postdoc at Carey Lab, we met briefly, a few weeks ago.

Megan told me you wanted somebody to give you feedback on the LocoMouse
code.

I am using Matlab R2015b and I'm getting started using the code. I found
some outdated commands that would cause errors and corrected them.

I also changed the use of vid.NumberOfFrames due to a warning that it
won't be used in future.

Finally I put in some lines that check if export_fig() exists. The error
message includes the gitHub link to it. :)

cheers

Dennis

DennisEckmeier committed:
Made it run in Matlab R2015b

Hi,

I'm the new postdoc at Carey Lab, we met briefly, a few weeks ago.

Megan told me you wanted somebody to give you feedback on the LocoMouse
code.

I am using Matlab R2015b and I'm getting started using the code. I found
some outdated commands that would cause errors and corrected them.

I also changed the use of vid.NumberOfFrames due to a warning that it
won't be used in future.

Finally I put in some lines that check if export_fig() exists. The error
message includes the gitHub link to it. :)

cheers

Dennis

joaofayad commented:
Hi Dennis,

I added my own account as a collaborator, so I think I should get the notifications now.

Thanks for testing the code and the changes you made. I had included some code for the newer behaviour of VideoReader when it comes to the vid.NumberOfFrames and other things. Did you check that? Its under tracking code\matlab2014. Also, did you have any problems reading a frame from the video inside a parfor loop? I  also had to change some of that, and I didn't see that in the changes you made.

I'm just not sure we should merge the changes as I don't know if they are backwards compatible. I know some the functions under tracking code\matlab2014 are not... That is why I have two versions of some functions. Perhaps you could commit your changes due to VideoReader only to the functions inside tracking code\matlab2014. If its not a lot of trouble, maybe you can check what works for all versions and what needs to be kept separate. It really sucks that VideoReader changed this much. 

The other changes I'm happy to merge.

Cheers,
Joao

DennisEckmeier commented:
Hi Joao,

I'm glad you found my changes useful. I did not have problems reading 
frames in the parfor, I didn't touch the newwindow function at all. The 
changes I made were sufficient to run it on Matlab 2015. As for the 
Matlab version, one could make an if clause to run different versions of 
the code on different Matlab versions (non elegant). But I am pretty 
sure one can also use VideoReader in a way that that works for 2014 and 
2015. I am actually pretty sure that my changes would work on 2014, too, 
but that should be tested, I guess ;).

I made one change to the newwindow function so far, that cuts out the sc 
function (and the custom function with the incredibly long name ;)) to 
adjust brightness and contrast in gray scale images. I replaced the 
function with two lines that seem to do the same thing (I compared 
images created with both codes, subtracted them from each other and got 
all zero matrices as a result). Unless something weird was happening in 
the background while I was running the tracking the first time, I think 
I cut processing time down by 20% for the example data with this change. 
It also gets rid of a dependency on somebody else's custom code, which I 
think is a plus. Of course I don't know the reason why you chose to use 
that function, so please consider it a mere suggestion on my side.

best wishes to Japan!

Dennis

On 2015-10-23 02:14, joaofayad wrote:

> Hi Dennis,
> 
> I added my own account as a collaborator, so I think I should get the
> notifications now.
> 
> Thanks for testing the code and the changes you made. I had included
> some code for the newer behaviour of VideoReader when it comes to the
> vid.NumberOfFrames and other things. Did you check that? Its under
> tracking code\matlab2014. Also, did you have any problems reading a
> frame from the video inside a parfor loop? I also had to change some
> of that, and I didn't see that in the changes you made.
> 
> I'm just not sure we should merge the changes as I don't know if they
> are backwards compatible. I know some the functions under tracking
> code\matlab2014 are not... That is why I have two versions of some
> functions. Perhaps you could commit your changes due to VideoReader
> only to the functions inside tracking code\matlab2014. If its not a
> lot of trouble, maybe you can check what works for all versions and
> what needs to be kept separate. It really sucks that VideoReader
> changed this much.
> 
> The other changes I'm happy to merge.
> 
> Cheers,
> Joao
> 
> ## 
> 
> Reply to this email directly or view it on GitHub [1].
> 
>   *
> 
> ## Links:
> 
> [1] https://github.com/careylab/LocoMouse/pull/1#issuecomment-150399584
